ext/ddtrace_profiling_native_extension/collectors_cpu_and_wall_time_worker.c

```diff
@@ -846,22 +877,43 @@ static VALUE _native_stats(DDTRACE_UNUSED VALUE self, VALUE instance) {
+    ID2SYM(rb_intern("cpu_sampling_time_ns_total")),                 /* => */ pretty_cpu_sampling_time_ns_total,
+    ID2SYM(rb_intern("cpu_sampling_time_ns_avg")),                   /* => */ pretty_cpu_sampling_time_ns_avg,
+    ID2SYM(rb_intern("cpu_sleeping_time_ns_avg")),                   /* => */ pretty_cpu_sleeping_time_ns_avg,
+    ID2SYM(rb_intern("allocation_sampling_time_ns_min")),            /* => */ pretty_allocation_sampling_time_ns_min,
+    ID2SYM(rb_intern("allocation_sampling_time_ns_max")),            /* => */ pretty_allocation_sampling_time_ns_max,
+    ID2SYM(rb_intern("allocation_sampling_time_ns_total")),          /* => */ pretty_allocation_sampling_time_ns_total,
+    ID2SYM(rb_intern("allocation_sampling_time_ns_avg")),            /* => */ pretty_allocation_sampling_time_ns_avg,
     ID2SYM(rb_intern("allocations_during_sample")),                  /* => */ UINT2NUM(state->stats.allocations_during_sample),
+    ID2SYM(rb_intern("allocation_sampler_event_rate")),              /* => */ DBL2NUM(discrete_dynamic_sampler_event_rate(state->allocation_sampler)),
+    ID2SYM(rb_intern("allocation_sampler_probability")),             /* => */ DBL2NUM(discrete_dynamic_sampler_probability(state->allocation_sampler)),
+    ID2SYM(rb_intern("allocation_sampler_time_ns")),                 /* => */ ULONG2NUM(discrete_dynamic_sampler_sampling_time_ns(state->allocation_sampler)),
+    ID2SYM(rb_intern("allocation_sampler_target_overhead_adj")),     /* => */ DBL2NUM(discrete_dynamic_sampler_target_overhead_adjustment(state->allocation_sampler)),
+
+    ID2SYM(rb_intern("cpu_sampled")),                                                /* => */ UINT2NUM(state->stats.cpu_sampled),
+    ID2SYM(rb_intern("skipped_cpu_sample_because_of_dynamic_sampling_rate")),        /* => */ UINT2NUM(state->stats.skipped_cpu_sample_because_of_dynamic_sampling_rate),
+    ID2SYM(rb_intern("effective_cpu_sample_rate")),                                  /* => */ effective_cpu_sample_rate,
+    ID2SYM(rb_intern("allocation_sampled")),                                         /* => */ ULONG2NUM(state->stats.allocation_sampled),
+    ID2SYM(rb_intern("skipped_allocation_sample_because_of_dynamic_sampling_rate")), /* => */ ULONG2NUM(state->stats.skipped_allocation_sample_because_of_dynamic_sampling_rate),
+    ID2SYM(rb_intern("effective_allocation_sample_rate")),                           /* => */ effective_allocation_sample_rate,
   };
   for (long unsigned int i = 0; i < VALUE_COUNT(arguments); i += 2) rb_hash_aset(stats_as_hash, arguments[i], arguments[i+1]);
   return stats_as_hash;
 }
 
+static VALUE _native_reset_stats(DDTRACE_UNUSED VALUE self, VALUE instance) {
+  struct cpu_and_wall_time_worker_state *state;
+  TypedData_Get_Struct(instance, struct cpu_and_wall_time_worker_state, &cpu_and_wall_time_worker_typed_data, state);
+  reset_stats(state);
+  return Qnil;
+}
```

Comment:
Yeah I'll leave this for the other PR that will actually export stats. In fact, I may actually move all the stats changes there since there's no easy way to consume or test them in this one.

Reply:
https://datadoghq.atlassian.net/browse/PROF-9036